libs/sr/ui/src/LightCone/LightConeEditor/LightConeEditor.tsx

```diff
@@ -0,0 +1,255 @@
+  const [dirtyDatabase, setDirtyDatabase] = useForceUpdate()
+
+  useEffect(
+    () => database.lightCones.followAny(setDirtyDatabase),
+    [database, setDirtyDatabase]
+  )
+
+  useEffect(() => {
+    if (lightConeIdToEdit === 'new') {
+      dispatchLightCone({ type: 'reset' })
+    }
+    const dbLightCone =
+      lightConeIdToEdit &&
+      dirtyDatabase &&
+      database.lightCones.get(lightConeIdToEdit)
+    if (dbLightCone) {
+      dispatchLightCone({
+        type: 'overwrite',
+        lightCone: deepClone(dbLightCone),
+      })
+    }
+  }, [lightConeIdToEdit, database, dirtyDatabase])
+
+  const [lightCone, dispatchLightCone] = useReducer(lightConeReducer, undefined)
+  const sheet: ILightConeSheet | undefined = lightCone
+    ? tempLightConeSheet
+    : undefined
+
+  const update = useCallback(
+    (newValue: Partial<ILightCone>) => {
+      // const newSheet = newValue.key ? getLightConeSheet(newValue.key) : sheet!
+      const newSheet = newValue.key ? tempLightConeSheet : sheet!
+
+      function pick<T>(
+        value: T | undefined,
+        available: readonly T[],
+        prefer?: T
+      ): T {
+        return value && available.includes(value)
+          ? value
+          : prefer ?? available[0]
+      }
+
+      if (newValue.key) {
+        newValue.superimpose = pick(
+          lightCone?.superimpose,
+          newSheet.superimpose,
+          Math.min(...newSheet.superimpose) as SuperimposeKey
+        )
+
+        // for some reason, location property not being added (undefined) if no character location is set, so force init
+        newValue.location = newValue.location ?? ''
+      }
+      if (newValue.level) {
+        newValue.level = clamp(newValue.level, 0, lightConeMaxLevel)
+      }
+
+      dispatchLightCone({ type: 'update', lightCone: newValue })
+    },
+    [lightCone, sheet, dispatchLightCone]
+  )
+
+  const reset = useCallback(() => {
+    dispatchLightCone({ type: 'reset' })
+  }, [])
+
+  useEffect(() => {
+    if (lightConeIdToEdit === 'new') {
+      dispatchLightCone({ type: 'reset' })
+    }
+  }, [lightConeIdToEdit])
+
+  return (
+    <Container>
+      <CardThemed bgt="dark">
+        <CardHeader title="Light Cone Editor" />
+        <CardContent sx={{ display: 'flex', flexDirection: 'column', gap: 1 }}>
+          <Grid container spacing={1} columns={{ xs: 1, md: 2 }}>
+            {/* name */}
+            <Grid item xs={1} md={true} display="flex">
+              <Select
+                value={lightCone?.key || ''}
+                sx={{ flexGrow: 1 }}
+                onChange={(e) =>
+                  update({ key: e.target.value as LightConeKey })
+                }
+              >
+                {allLightConeKeys.map((lc) => (
+                  <MenuItem key={lc} value={lc}>
+                    {lc}
+                  </MenuItem>
+                ))}
+              </Select>
+            </Grid>
+
+            {/* superimpose */}
```

Comment:
I think everything after this should be disabled if the lightcone object is undefined